### DISCUSSION TITLE
Response Body Returning String when using a response macro

### DISCUSSION THREAD

**Discussion by SimonBroekaert** [dedoc/scramble#203] [Ideas]
Hi! Awesome tool so far!

Is there a way to make macro's work automatically?

![image](https://github.com/dedoc/scramble/assets/35606498/b4f84b11-eda6-4d8f-a5a8-bfa5e40e5242)

I have these macros and use them like this:

![image](https://github.com/dedoc/scramble/assets/35606498/d3d74e67-a1c3-4499-bff2-9b02c4e17eed)

Not sure if this can ever work, but it would be amazing! Thanks in advance

**Comment by romalytvynenko** [comment]
@SimonBroekaert Wow, is that a documented feature?

In Laravel documentation I found that when you define macro, you can call it on `ResponseFactory` (https://laravel.com/docs/10.x/responses#response-macros). Essentially, something like this:
```php
return response()->apiSuccessResponse(...);
```

But I never thought it is possible to call them just like functions 0_0

**Reply by SimonBroekaert** [reply]
Oh sorry @romalytvynenko , I forgot to show my helpers.php file! Sorry for the confusion haha!

```php
if (! function_exists('apiSuccessResponse')) {
    function apiSuccessResponse(string $message = null, JsonResource|Collection|array $data = [], int $status = 200): JsonResponse
    {
        return response()->apiSuccess($message, $data, $status);
    }
}

if (! function_exists('apiErrorResponse')) {
    function apiErrorResponse(string $message = null, JsonResource|Collection|array $errors = [], int $status = 400): JsonResponse
    {
        return response()->apiError($message, $errors, $status);
    }
}
```

So it is like you found in the documentation, but with this helpers.php file that gets autoloaded in composer.json like this: ...

```json
{
    "autoload": {
        "files": [
            "app/Helpers/helpers.php"
        ]
    },
}
```

... I can then call them like functions.
